sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/StorageImplUtils.java

```diff
@@ -297,4 +301,53 @@ public static void copyToOutputStream(InputStream source, long writeLength, Outp
             count = source.read(retrievedBuff, 0, nextCopy);
         }
     }
+
+    /**
+     * Logs the string to sign if a valid context is provided.
+     *
+     * @param logger {@link ClientLogger}
+     * @param stringToSign The string to sign to log.
+     * @param context Additional context to determine if the string to sign should be logged.
+     */
+    public static void logStringToSign(ClientLogger logger, String stringToSign, Context context) {
+        if (context != null && Boolean.TRUE.equals(context.getData(Constants.STORAGE_LOG_STRING_TO_SIGN).orElse(false))) {
+            logger.info("The string to sign computed by the SDK is: {}{}", stringToSign,
+                System.lineSeparator());
+            logger.warning("Please remember to disable '{}' before going to production as this "
+                + "string can potentially contain PII.", Constants.STORAGE_LOG_STRING_TO_SIGN);
+        }
+    }
+
+    /**
+     * Converts the storage exception message.
+     *
+     * @param message The storage exception message
+     * @param response The storage service response.
+     * @return The converted storage exception message.
+     */
+    public static String convertStorageExceptionMessage(String message, HttpResponse response) {
+        if (response != null) {
+            if (response.getStatusCode() == 403) {
+                return String.format("If you are using a StorageSharedKeyCredential, and the server returned an "
```

Comment:
Let's make this message a constant as all parameters for the formatting are constant, that way we don't need to make additional calls to `String.format` and wait for this to be optimized by JIT